### DISCUSSION TITLE
Wait or sleep function for steady state?

### DISCUSSION THREAD

**Discussion by kemjensak** [isaac-sim/IsaacLab#380] [Q&A] [answered]
Hi! 
I'm making manipulation task that has many objects on table
All objects poses are randomized every episode
So I have to wait few seconds for objects to stop on the table.
Any function/method OR suggestion for this?
Thanks a lot for orbit framework!
![Screenshot from 2024-04-15 08-07-10](https://github.com/NVIDIA-Omniverse/orbit/assets/60342784/11921f07-5c1e-4d75-bf13-62d2795ca7ae)

**Comment by Mayankm96** [comment, marked as answer]
Do you have asynchronous termination, i.e., different environments that are expected to reset at different time steps? If so, then this isn't possible. In the other case (where all environments execute to their full episode length), you can add multiple steps to let them all "rest".

Alternately, what I do is: spawn a large number of times. Store all the poses in a database and sample from that database. It works fine for most of my applications.

**Reply by kemjensak** [reply]
Thank you for answer! I'll try both of them!
